core/src/main/java/edu/ucr/cs/riple/core/metadata/index/Index.java

```diff
@@ -82,26 +83,29 @@ public Index(Path path, Index.Type type, Factory<T> factory) {
   /** Starts the reading and index process. */
   public void index() {
     items.clear();
-    try (BufferedReader br = Files.newBufferedReader(this.path, UTF_8)) {
-      String line = br.readLine();
-      if (line != null) {
-        line = br.readLine();
-      }
-      while (line != null) {
-        T item = factory.build(line.split("\t"));
-        total++;
-        int hash;
-        if (type.equals(Index.Type.BY_CLASS)) {
-          hash = Objects.hash(item.encClass());
-        } else {
-          hash = Objects.hash(item.encClass(), item.encMethod());
-        }
-        items.put(hash, item);
-        line = br.readLine();
-      }
-    } catch (IOException e) {
-      e.printStackTrace();
-    }
+    paths.forEach(
+        path -> {
+          try (BufferedReader br = Files.newBufferedReader(path, UTF_8)) {
+            String line = br.readLine();
+            if (line != null) {
+              line = br.readLine();
+            }
+            while (line != null) {
+              T item = factory.build(line.split("\t"));
+              total++;
```

Comment:
Make `total` a private field. And add a getter.  You don't want to be accidentally updating the field anywhere :)

Reply:
4695aa7091841e051b1d0704fd960201d6457807